student.rb

```diff
@@ -17,4 +17,28 @@ def add_classroom(classroom)
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def tohash
+    temp_hash = {
+      @id => {
+        'age' => @age,
+        'name' => @name,
+        'p_p' => @parent_permission,
+        'id' => @id,
+        'classroom' => @classroom,
+        'rentals' => @rentals
+      }
+    }
+    tojson(temp_hash)
+  end
+
+  def tojson(temp_hash)
+    if File.zero?('persons.json')
+      File.write('persons.json', JSON.pretty_generate([temp_hash]))
+    else
+      data_from_json = JSON.parse(File.read('persons.json'))
+      data_from_json = [data_from_json] if data_from_json.class != Array
+      File.write('persons.json', JSON.pretty_generate(data_from_json << temp_hash))
+    end
+  end
```

Comment:
- [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20exit%20from%20your%20app%20save%20all%20the%20data%20(books%2C%20people%2C%20and%20rentals)%20into%20JSON%20files.), Please save the person data on exit only.